src/ed_object_models/spawn_sdf_in_gazebo.py

```diff
@@ -0,0 +1,78 @@
+#! /usr/bin/python
+
+import rospy
+import yaml
+import os
+import glob
+
+from gazebo_msgs.srv import SpawnModel
+from geometry_msgs.msg import Pose, Point, Quaternion
+from tf.transformations import quaternion_from_euler
+
+def from_yaml(yaml_path):
+    '''
+    Spawns a list of sdf files from a yaml file into Gazebo.
+    :param yaml_path: path to a yaml file.
+    :type yaml_path: str
+    :return: Spawns the sdf models.
+    '''
+
+    # Initialize ROS node
+    rospy.init_node('gazebo_simulator_object_spawner')
+
+    # Wait until gazebo is ready to spawn sdf models
+    rospy.wait_for_service('gazebo/spawn_sdf_model')
+    spawn_model_prox = rospy.ServiceProxy('gazebo/spawn_sdf_model', SpawnModel)
+
+    # Load yaml with list of objects that need to be loaded
+    if not os.path.isfile(yaml_path):   # Check if yaml_path is a path to a file.
+        print('Warning: Could not find ' + yaml_path)
+        return
+
+    with open(yaml_path, 'r') as f:
+        items = yaml.safe_load(f)
+
+    # Get paths in $GAZEBO_MODEL_PATH
+    model_paths = os.environ['GAZEBO_MODEL_PATH'].split(os.pathsep)
+
+    # Iterate over objects and spawn
+    for item in items:
+        # Define object pose
+        object_pose = Pose()
+        object_pose.position = Point(item['x'], item['y'], item['z'])
+        object_pose.orientation = Quaternion(*quaternion_from_euler(item.get('roll', 0),
```

Comment:
Why create a quaternion from a quaternion?

Reply:
The function quaternion_from_euler creates a quaternion q from euler angles. I can't pass this quaternion directly to object_pose.orientation, since the orientation is defined as object_pose.orientation.x etc. So I use Quaternion to do this for me, where Quaternion(*q) maps the four elements of q to the four inputs of Quaternion.

Reply:
@MatthijsBurgh Do you still require me to make changes?